File: back-end/subscriptionsWS/routes/membersRouter.js
const express = require("express");
const router = express.Router();

const membersBLL = require("../BLL/membersBll");

router.get("/", async (req, res) => {
  try {
    const members = await membersBLL.getALLMembers();
    res.status(200).json(members);
  } catch (e) {
    res.status(500).json(e);
  }
});

router.get("/:id", async (req, res) => {
  try {
    const { id } = req.params;
    const member = await membersBLL.getOneMember(id);
    res.status(200).json(member);
  } catch (e) {
    res.status(500).json(e);
  }
});

router.post("/", async (req, res) => {
  try {
    const obj = req.body;
    const result = await membersBLL.createMember(obj);
    res.status(200).json(result);
  } catch (e) {
    res.status(500).json(e);
  }
});

router.put("/:id", async (req, res) => {
  try {
    const { id } = req.params;
    const obj = req.body;
    const result = await membersBLL.updateMember(id,obj);
    res.status(200).json(result);
  } catch (e) {
    res.status(500).json(e);
  }
});

router.delete("/:id", async (req, res) => {
  try {
    const { id } = req.params;
    const result = await membersBLL.deleteMember(id);
    res.status(200).json(result);
  } catch (e) {
    res.status(500).json(e);
  }
});

module.exports = router;

File: back-end/subWS/routers/membersRouter.py
from flask import Blueprint, jsonify, request
from Bll.mongoDbBll import MongoDbBll


members_bll = MongoDbBll("members")

members = Blueprint("members", __name__)


@members.route("/", methods=["GET"])
def get_all_members():
    members = members_bll.get_all_data()
    return jsonify(members)


@members.route("/<string:id>", methods=["GET"])
def get_member(id):
    member = members_bll.get_one_data(id)
    return jsonify(member)


@members.route("/", methods=["POST"])
def add_member():
    obj = request.json
    status = members_bll.add_data(obj)
    return jsonify(status)


@members.route("/<string:id>", methods=["PUT"])
def update_member(id):
    obj = request.json
    obj.pop('_id', None)
    status = members_bll.update_data(id, obj)
    return jsonify(status)


@members.route("/<string:id>", methods=["DELETE"])
def delete_member(id):
    status = members_bll.delete_data(id)
    return jsonify(status)
